### PR TITLE
New/dynamic routes

### LINKED ISSUE #234: New Play : Dynamic Routes
Thank You! 

## What's the Play about?
 its all about dynamic routes using react router v6. Wher in user pass the dynamic data as a parameter and based on the parameter app shows the relevant output. 

## What ReactJS concept will be used to create this play?
react-router-v6, useParam hook 

## Are you willing to take it up for implementation?
yes

### PR DESCRIPTION
**Pull request by Deepak8717**
> First thing, PLEASE READ THIS: [ReactPlay Code Review Checklist](https://github.com/atapas/react-play/wiki/ReactPlay-Code-Review-Checklist)

# Description

A new projects to learn about dynamic routes 
a beginner friendly app where you put the meal category in the url and based on the input app shows the relevant recipes.  

closes #234 
## Type of change

Please delete options that are not relevant.

- [ ] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update

# How Has This Been Tested?

have manually tested each and every functionality along with warnings and errors.

# Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings

### CODE REVIEW

**Comment by atapas** [comment]
@Deepak8717 Create an issue explaining what is the objective.

**Comment by atapas** [comment]
@Deepak8717 Is it review ready?

**Comment by Deepak8717** [comment]
@atapas yes

File: src/plays/dynamic-routes/Data.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+const data = [
+  {
+    name: "Omelette",
```
Comment (atapas) [inline src/plays/dynamic-routes/Data.js:3]: Adding some images will make the play look attarctive.

**Review by atapas — COMMENTED** [review]
In the initial state it is not easy to understand what is selected.

image

File: src/plays/dynamic-routes/Readme.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+## Creator Information
+
+User: Deepak8717
+Gihub Link: https://github.com/Deepak8717
+Blog:
+Video:
+
+## Implementation Details
+
+to get the url input paramete value we need to tweak the path of the this project in the play-mets.js file
```
Comment (atapas) [inline src/plays/dynamic-routes/Readme.md:19]: typo play-meta

**Review by atapas — CHANGES_REQUESTED** [review]

**Comment by Deepak8717** [comment]
@atapas Please check. I have added the images, fixed the responsiveness, fixed initial menu selection bug  and updated readme file.

**Comment by atapas** [comment]
@Deepak8717 A small one

image

**Comment by Deepak8717** [comment]
@atapas done

File: src/plays/dynamic-routes/Readme.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+## Creator Information
+
+User: Deepak8717
+Gihub Link: https://github.com/Deepak8717
+Blog:
+Video:
+
+## Implementation Details
+
+to get the url input parameter value we need to tweak the path of the this project in the play-metas.js file
```
Comment (atapas) [inline src/plays/dynamic-routes/Readme.md:19]: it is play-meta.js

File: src/plays/dynamic-routes/Readme.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+from ""/plays/dynamic-routes/" to "/plays/dynamic-routes/:menu"
```
Comment (atapas) [inline src/plays/dynamic-routes/Readme.md:20]: Why two ""?

File: src/plays/dynamic-routes/Readme.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+we need to use useParams() hook to get the input value of url and pass to function that filter the data so we can fetch the recipes based input value. if our input matches the meal categories. we show the data on screen.
+
+## Consideration
+
+Update all considerations(if any)
+
+## Resources
+
```
Comment (atapas) [inline src/plays/dynamic-routes/Readme.md:28]: Can add links that helped you learn about routing and params.

**Review by atapas — CHANGES_REQUESTED** [review]

File: src/plays/dynamic-routes/Readme.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+## Creator Information
+
+User: Deepak8717
+Gihub Link: https://github.com/Deepak8717
+Blog:
+Video:
+
+## Implementation Details
+
+to get the url input parameter value we need to tweak the path of the this project in the play-metas.js file to "/plays/dynamic-routes/:menu"
```
Comment (atapas) [inline src/plays/dynamic-routes/Readme.md:19]: @Deepak8717 buddy... it is not `play-metas.js`

**Review by atapas — CHANGES_REQUESTED** [review]

**Comment by Deepak8717** [comment]
@atapas fixed it

**Review by atapas — APPROVED** [review]

**Pull request merged**